engines/alcachofa/alcachofa.h

```diff
@@ -0,0 +1,199 @@
+#include "alcachofa/input.h"
+#include "alcachofa/sounds.h"
+#include "alcachofa/player.h"
+#include "alcachofa/scheduler.h"
+#include "alcachofa/console.h"
+#include "alcachofa/game.h"
+
+namespace Alcachofa {
+
+class IDebugHandler;
+class IRenderer;
+class DrawQueue;
+class World;
+class Script;
+class GlobalUI;
+class Menu;
+class Game;
+struct AlcachofaGameDescription;
+
+constexpr int16 kSmallThumbnailWidth = 160; // for ScummVM
+constexpr int16 kSmallThumbnailHeight = 120;
+static constexpr int16 kBigThumbnailWidth = 341; // for in-game
+static constexpr int16 kBigThumbnailHeight = 256;
+
+
+enum class SaveVersion : Common::Serializer::Version {
+	Initial = 0
+};
+static constexpr SaveVersion kCurrentSaveVersion = SaveVersion::Initial;
+
+class MySerializer : public Common::Serializer {
+public:
+	using Common::Serializer::Serializer;
+
+	Common::SeekableReadStream &readStream() {
+		assert(isLoading() && _loadStream != nullptr);
+		return *_loadStream;
+	}
+
+	Common::WriteStream &writeStream() {
+		assert(isSaving() && _saveStream != nullptr);
+		return *_saveStream;
+	}
+};
+
+class Config {
+public:
+	Config();
+
+	inline bool &subtitles() { return _subtitles; }
+	inline bool &highQuality() { return _highQuality; }
+	inline bool &bits32() { return _bits32; }
+	inline uint8 &musicVolume() { return _musicVolume; }
+	inline uint8 &speechVolume() { return _speechVolume; }
+
+	void loadFromScummVM();
+	void saveToScummVM();
+
+private:
+	bool
+		_subtitles = true,
+		_highQuality = true,
+		_bits32 = true;
+	uint8
+		_musicVolume = 255,
+		_speechVolume = 255;
+};
+
+class AlcachofaEngine : public Engine {
+protected:
+	// Engine APIs
+	Common::Error run() override;
+public:
+	AlcachofaEngine(OSystem *syst, const ADGameDescription *gameDesc);
+	~AlcachofaEngine() override;
+
+	inline const ADGameDescription &gameDescription() const { return *_gameDescription; }
+	inline IRenderer &renderer() { return *_renderer; }
+	inline DrawQueue &drawQueue() { return *_drawQueue; }
+	inline Camera &camera() { return _camera; }
+	inline Input &input() { return _input; }
+	inline Sounds &sounds() { return _sounds; }
+	inline Player &player() { return *_player; }
+	inline World &world() { return *_world; }
+	inline Script &script() { return *_script; }
+	inline GlobalUI &globalUI() { return *_globalUI; }
+	inline Menu &menu() { return *_menu; }
+	inline Scheduler &scheduler() { return _scheduler; }
+	inline Console &console() { return *_console; }
+	inline Game &game() { return *_game; }
+	inline Config &config() { return _config; }
+	inline bool isDebugModeActive() const { return _debugHandler != nullptr; }
+
+	uint32 getMillis() const;
+	void setMillis(uint32 newMillis);
+	void pauseEngineIntern(bool pause) override;
+	void playVideo(int32 videoId);
+	void fadeExit();
+	void setDebugMode(DebugMode debugMode, int32 param);
+
+	uint32 getFeatures() const;
+	Common::String getGameId() const;
+
+	bool hasFeature(EngineFeature f) const override {
+		return
```

Comment:
I recommend implementind extended saves right from the beginning

Reply:
Isn't the extended saves a metaengine feature and not an engine feature? In the metaengine I use the common `AdvancedMetaEngineBase::checkExtendedSaves` and have tested extended saves already working.

If I am missing the correct engine feature to declare here, please let me know